phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1570,6 +1575,15 @@ public GetVersionResponse call(MetaDataService instance)
                     systemCatalogTimestamp = systemCatalogTimestamp < versionResponse.getSystemCatalogTimestamp() ?
                       systemCatalogTimestamp: versionResponse.getSystemCatalogTimestamp();
                 }
+
+                if (compatibility.getErrorCode() != 0) {
+                    if (compatibility.getErrorCode() == SQLExceptionCode.OUTDATED_JARS.getErrorCode()) {
+                        errorMessage.setLength(errorMessage.length()-1);
```

Comment:
Do we need to setLength on the StringBuilder object? We can just directly call toString right?

Reply:
The setLength(length-1) gets rid of a trailing semicolon if there are multiple outdated jars error messages strung together (for loop at line 1540)

Reply:
If there aren't multiple messages, and the loop runs just once, does it print the correct thing still or leave out the last character?

Reply:
@christinefeng  ping on this one^

Reply:
If it runs just once, there's still that trailing semicolon from line 1559, so the set length-1 gets rid of that semicolon
Not sure if I should add something similar to the incompatible_client_server_jar error message in case of multiple messages?

Reply:
That's more for cases when we have SYSCAT on more than 1 region. Your change should be fine.